src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -48,13 +33,15 @@ public class RazorPageActivator : IRazorPageActivator
             HtmlEncoder htmlEncoder,
             IModelExpressionProvider modelExpressionProvider)
         {
-            _activationInfo = new ConcurrentDictionary<Type, PageActivationInfo>();
-            _metadataProvider = metadataProvider;
-            _urlHelperAccessor = context => urlHelperFactory.GetUrlHelper(context);
-            _jsonHelperAccessor = context => jsonHelper;
-            _diagnosticSourceAccessor = context => diagnosticSource;
-            _htmlEncoderAccessor = context => htmlEncoder;
-            _modelExpressionProviderAccessor = context => modelExpressionProvider;
+            _razorPagePropertyActivator = new RazorPagePropertyActivator(
```

Comment:
Yup, primarily the reason why I did not make `RazorPagePropertyActivator` a service that gets injected. If we decide to simplify it to be just be a VDDActivator, it we might get away with a lot less change here. .